src/rules/azureResourceGraph.ts

```diff
@@ -77,35 +108,161 @@ export class ResourceGraphRule implements BaseRule<ResourceGraphTarget> {
       return `${initalTable} ${groupQuery} ${queryEnding}`;
     }
   }
 
-  toScanResult(response: ResourceGraphModels.ResourcesResponse): ScanResult {
+  async sendRequest(
+    target: ResourceGraphTarget,
+    resourceId: string,
+    request: RequestEvaluationObject
+  ) {
+    if (!isRequestEvaluation(this.evaluation)) {
+      throw Error('A valid request evaluation was not found');
+    }
+    const token = await target.credential.getToken(
+      'https://graph.microsoft.com/.default'
+    );
+    const resourceManagementClient = await this.getResourceManagmentClient(
+      resourceId
+    );
+    const options = {
+      url: await this.getRequestUrl(
+        resourceId,
+        resourceManagementClient,
+        request
+      ),
+      method: request.httpMethod as HttpMethods,
+      headers: {
+        Authorization: `Bearer ${token?.token}`,
+        'Content-Type': 'application/json',
+      },
+    };
+    let response = await resourceManagementClient.sendRequest(options);
+    // if the response returns an error because the default api verison is invalid, then parse the error message to retrieve a valid one and try again
+    if (
+      response.status === 400 &&
+      response.parsedBody.error.code === 'NoRegisteredProviderFound'
+    ) {
+      const apiVersion = this.getApiVersionFromError(
+        response.parsedBody.error.message
+      );
+      options.url = await this.getRequestUrl(
+        resourceId,
+        resourceManagementClient,
+        request,
+        apiVersion
+      );
+      response = await resourceManagementClient.sendRequest(options);
+    }
+    return response;
+  }
+
+  async getResourceManagmentClient(resourceId: string) {
+    const subscriptionId = this.getElementFromId('subscription', resourceId);
+    return await new ResourceManagementClient(
+      new AzureIdentityCredentialAdapter(new DefaultAzureCredential()),
+      subscriptionId
+    );
+  }
+
+  getApiVersionFromError(errorMessage: string) {
+    // splits message so the initial api version that failed is not included in the regex match.
+    const splitMessage = errorMessage.split('The supported api-versions are ');
+    const versions = splitMessage[1];
+    const apiVersions = versions.match(/\d\d\d\d-\d\d-\d\d(-preview)?/g);
+    if (apiVersions) {
+      return apiVersions.sort()[apiVersions.length - 1];
+    }
+    throw Error('Unable to find a valid api version');
+  }
+
+  async getDefaultApiVersion(
+    resourceId: string,
+    client: ResourceManagementClient
+  ) {
+    const provider = this.getElementFromId('provider', resourceId);
+    const resourceType = this.getElementFromId('resourceType', resourceId);
+    const providerResponse = await client.providers.get(provider);
+    const apiVersion = providerResponse.resourceTypes?.find(
+      r => r.resourceType === resourceType
+    )?.defaultApiVersion;
+    return apiVersion;
+  }
+
+  getElementFromId(
```

Comment:
Take a look at named capture groups. I think it's a cleaner way of getting the info you want here

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Groups_and_Ranges#using_named_groups

Ex:

```javascript
const id = '/subscriptions/1234/resourceGroups/foo/providers/Microsoft.Storage/storageAccounts/bar';
const regex = /\/subscriptions\/(?<subscriptionId>.*)\/resourceGroups\/(?<resourceGroup>.*)\/providers\/(?<provider>.*)\/(?<resourceType>.*)\/(?<name>.*)/;
const match = regex.exec(id);

match.groups.subscriptionId
match.groups['name']
```